packages/android_alarm_manager/android/src/main/java/io/flutter/plugins/androidalarmmanager/AlarmService.java

```diff
@@ -15,27 +15,61 @@
 import android.util.Log;
 import io.flutter.app.FlutterActivity;
 import io.flutter.app.FlutterApplication;
+import io.flutter.plugin.common.MethodChannel;
 import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback;
+import io.flutter.view.FlutterCallbackInformation;
 import io.flutter.view.FlutterMain;
 import io.flutter.view.FlutterNativeView;
+import io.flutter.view.FlutterRunArguments;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 public class AlarmService extends Service {
   public static final String TAG = "AlarmService";
+  private static AtomicBoolean sStarted = new AtomicBoolean(false);
   private static FlutterNativeView sSharedFlutterView;
+  private static MethodChannel sBackgroundChannel;
   private static PluginRegistrantCallback sPluginRegistrantCallback;
 
-  private FlutterNativeView mFlutterView;
-  private String appBundlePath;
+  private String mAppBundlePath;
+
+  public static void onInitialized() {
+    sStarted.set(true);
+  }
+
+  public static void startAlarmService(Context context, long callbackHandle) {
```

Comment:
Some comments about what is going on here will probably be helpful to 3p developers trying to write their own plugins.